### PR TITLE
[share] add admin permission

### PR DESCRIPTION
**Pull request by zMingGit**

### CODE REVIEW

File: seahub/share/models.py [hunk 1]
```diff
@@ -150,6 +152,79 @@ def get_dir_link_by_path(self, username, repo_id, path):
...
+
+    def get_repos_with_admin_share_to(self, username):
+        """Get repo id with the admin permission record.
+        """
+        shared_repos = super(ExtraSharePermissionManager, self).filter(
+            share_to=username, permission='admin'
+        )
+        return [e.repo_id for e in shared_repos]
+
+    def get_users_by_repo_and_admin(self, repo_id):
```
Comment (freeplant) [inline seahub/share/models.py]: get_users_by_repo_and_admin -> get_admin_users_by_repo

File: seahub/share/models.py [hunk 2]
```diff
@@ -150,6 +152,79 @@ def get_dir_link_by_path(self, username, repo_id, path):
     def get_valid_dir_link_by_token(self, token):
         return self._get_valid_file_share_by_token(token)
 
+
+class ExtraSharePermissionManager(models.Manager):
+    def get_user_permission(self, repo_id, username):
+        """Get user permission  in Library.
```
Comment (freeplant) [inline seahub/share/models.py]: Get a user's permission of a library

File: seahub/share/models.py [hunk 3]
```diff
@@ -150,6 +152,79 @@ def get_dir_link_by_path(self, username, repo_id, path):
...
+        """
+        record_list = super(ExtraSharePermissionManager, self).filter(
+            repo_id=repo_id, share_to=username
+        )
+        if len(record_list) > 0:
+            return record_list[0].permission
+        else:
+            return None
+
+    def get_repos_with_admin_share_to(self, username):
```
Comment (freeplant) [inline seahub/share/models.py]: get_repos_with_admin_share_to -> get_repos_with_admin_permission

File: seahub/share/models.py [hunk 4]
```diff
@@ -150,6 +152,79 @@ def get_dir_link_by_path(self, username, repo_id, path):
...
+        """Get repo id with the admin permission record.
```
Comment (freeplant) [inline seahub/share/models.py]: Get repo id with the admin permission record.  -> Get repo id list a user has admin permission.

File: seahub/constants.py [hunk 5]
```diff
@@ -4,3 +4,8 @@
 
 # Guest user have limited operations, can not create group and library.
 GUEST_USER = 'guest'
+
+# Permissions
+PERMISSION_READY = 'r'
+PERMISSION_READY_WRITE = 'rw'
```
Comment (llj) [inline seahub/constants.py]: 上面两行中的 `READY` 应该是 `READ` 吧？
`PERMISSION_READ_ONLY = 'r'`

**Pull request merged**